modules/UI/videolayout/LocalVideo.js

```diff
@@ -112,8 +112,11 @@ LocalVideo.prototype.changeVideo = function (stream) {
         localVideoContainer.removeChild(localVideo);
         // when removing only the video element and we are on stage
         // update the stage
-        if(this.isCurrentlyOnLargeVideo())
-            this.VideoLayout.updateLargeVideo(this.id);
+        if (this.isCurrentlyOnLargeVideo()) {
+            this.VideoLayout.updateLargeVideo(
+                this.id,
+                true /* force - stream removed for the same user ID */);
+        }
```

Comment:
With your latest changes, could the if statements for checking isCurrentlyOnLargeVideo be removed from these updateLargeVideo calls? It seems like they could be so that updateLargeVideo is called blindly.